### PR TITLE
Don't show a signup url for MAAS

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3552.

### CODE REVIEW

**Comment by huwshimi** [comment]
[Don't show creds signup url for MAAS](https://trello.com/c/oGDsJRlX/352-dont-show-creds-signup-url-for-maas)

**Comment by huwshimi** [comment]
QA:
- use the GUI with MAAS
- open your profile and open the form to add a credential
- you should not see a signup url

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8537/

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js [hunk 1]
```diff
@@ -76,6 +76,22 @@ describe('DeploymentCredentialAdd', function() {
...
       message: 'a message'
     });
+    getCloudProviderDetails.withArgs('maas').returns({
+      id: 'maas',
+      showLogo: false,
+      title: 'MAAS',
+      forms: {
+        'access-key': [{
+          id: 'access-key',
+          title: 'The EC2 access key'
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js]: Even if this is just used to set up a mock, it's weird to see ec2 stuff on a maas section.

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js [hunk 2]
```diff
@@ -1125,4 +1141,11 @@ describe('DeploymentCredentialAdd', function() {
       </div>);
     expect(output).toEqualJSX(expected);
   });
+
+  it('does not show a signup url if the cloud does not have one', () => {
+    const comp = renderComponent({
+      cloud: {name: 'maas', cloudType: 'maas'}
+    });
+    assert.equal(comp.output.props.children[1], null);
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js]: strictEqual ?

**Comment by mitechie** [comment]
QA ok on MAAS. In thinking this through I'm also wondering about other providers such as the manual provider and lxd.

**Comment by mitechie** [comment]
Sorry, looking at the code is checks if a signup url is set so ignore me.

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js (hunk 1 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js]: Oops, thanks.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8539/

**Comment by huwshimi** [comment]
Thanks @frankban and @mitechie :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**